js/src/personal/kazi/Kazi.page.tsx

```diff
@@ -1,18 +1,29 @@
-import { Title, Text } from '@mantine/core'
+import { Title, Text, useMantineColorScheme } from '@mantine/core'
 import React, { useState } from 'react'
 import { FaSun, FaMoon } from 'react-icons/fa'
 import styles from './Kazi.module.css'
 
 export function KaziPage() {
   const [sun, setSun] = useState(true)
-  const [sunAnimate, setSunAnimate] = useState(false)
+  const [animate, setAnimate] = useState('')
+  const { setColorScheme } = useMantineColorScheme()
 
-  const sunToMoon = () => {
-    setSunAnimate(true)
-    setTimeout(() => {
-      setSun(!sun)
-      setSunAnimate(false)
-    }, 2000)
+  const changeIcon = () => {
+    if (animate === '') {
+      setAnimate(sun ? 'exitSun' : 'exitMoon')
```

Comment:
Haven't really done animations before in React, but this seems like so many sets to do a transition.
Could let this through though.

Reply:
When I was testing this manually what would happen before I added this logic was that when the animation ran for whichever icon moving to the left,  the subsequent icon would then also move to the left and then recenter itself in the middle with a snap animation.

Reply:
Looking into useTransition, it seems like it's more for highly resource-intensive tasks rather than smaller animations like changing the color scheme. Will definitely think about using this for my contact section after a form submission.